tx.go

```diff
@@ -91,12 +91,20 @@ func NewBatchTransaction(creator *skademlia.Keypair, tags []byte, payloads [][]b
 	return NewTransaction(creator, sys.TagBatch, append([]byte{byte(len(tags))}, buf...))
 }
 
-func AttachSenderToTransaction(sender *skademlia.Keypair, tx Transaction, parents ...*Transaction) Transaction {
+func AttachSenderToTransaction(sender *skademlia.Keypair, _tx *Transaction, parents ...*Transaction) *Transaction {
+	tx := *_tx
+	AttachSenderToTransactionInPlace(sender, &tx, parents...)
+	return &tx
+}
+
+func AttachSenderToTransactionInPlace(sender *skademlia.Keypair, tx *Transaction, parents ...*Transaction) {
```

Comment:
Rename to `AppendSenderToTransaction` to imply that it in-place modifies the transaction provided. Also please provide documentation on these functions.

Reply:
I don't think Append and Attach really imply this difference.

Reply:
The `Append` prefix pattern, I am taking it as a style guideline from the `fasthttp`, `fastjson` etc libraries that provide functions which modify some pointer in-place.